src/Microsoft.Performance.Toolkit.Engine/EngineExecutionResults.cs

```diff
@@ -19,6 +18,7 @@ namespace Microsoft.Performance.Toolkit.Engine
     ///     Represents the results of processing data using the <see cref="Engine"/>.
     /// </summary>
     public sealed class RuntimeExecutionResults
+        : ICookedDataRetrieval
```

Comment:
was this class already implementing this interface implicitly? I don't see anything added to the class.

Why are "results" able to retrieve data? I feel like that should be happening directly through the engine/runtime, not through its final result. but that might be a bigger change

Reply:
This class was already implementing the interface, yes. Adding the interface allows the interface extensions to work with this class.

The results are how clients of the engine retrieve any data from the output. @zachnew-msft designed it this way, and it seems to work well. But if you have some thoughts on how to improve it, happy to chat.